[cupertino_http] 3.0.0 Building Android faild

The error:
```
Unhandled exception:
ProcessException: Full command string: '/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-linux-android29 --sysroot=/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/ -fPIC -O3 -Wl,-z,max-page-size=16384 -fobjc-arc -DRELEASE -DNDEBUG /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/src/native_cupertino_bindings.m -framework Foundation --shared -o /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/libcupertino_http.so -Wl,-rpath,$ORIGIN -L/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/'.
Exit code: '1'.
For the output of the process check the logger output.
  Command: /Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-linux-android29 --sysroot=/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/ -fPIC -O3 -Wl,-z,max-page-size=16384 -fobjc-arc -DRELEASE -DNDEBUG /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/src/native_cupertino_bindings.m -framework Foundation --shared -o /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/libcupertino_http.so -Wl,-rpath,$ORIGIN -L/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/
#0      runProcess (package:native_toolchain_c/src/utils/run_process.dart:85:5)
<asynchronous suspension>
#1      RunCBuilder._compile (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:230:5)
<asynchronous suspension>
#2      RunCBuilder.runClangLike (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:202:7)
<asynchronous suspension>
#3      RunCBuilder.run (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:124:7)
<asynchronous suspension>
#4      CBuilder.run (package:native_toolchain_c/src/cbuilder/cbuilder.dart:205:5)
<asynchronous suspension>
#5      build (package:hooks/src/api/build_and_link.dart:191:5)
<asynchronous suspension>
#6      main (file:///Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/hook/build.dart:10:3)
<asynchronous suspension>

  Building assets for package:cupertino_http failed.
  build.dart returned with exit code: 255.
  To reproduce run:
  (cd /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/; /Users/a/A/PROGRAMS/flutter/bin/cache/dart-sdk/bin/dart --packages=/Users/a/A/PROJECTS/b/app/.dart_tool/package_config.json /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/cupertino_http/aece7ab44c/hook.dill --config=/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/cupertino_http/aece7ab44c/input.json )
  stderr:
  Unhandled exception:
ProcessException: Full command string: '/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-linux-android29 --sysroot=/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/ -fPIC -O3 -Wl,-z,max-page-size=16384 -fobjc-arc -DRELEASE -DNDEBUG /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/src/native_cupertino_bindings.m -framework Foundation --shared -o /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/libcupertino_http.so -Wl,-rpath,$ORIGIN -L/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/'.
Exit code: '1'.
For the output of the process check the logger output.
  Command: /Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-linux-android29 --sysroot=/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/ -fPIC -O3 -Wl,-z,max-page-size=16384 -fobjc-arc -DRELEASE -DNDEBUG /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/src/native_cupertino_bindings.m -framework Foundation --shared -o /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/libcupertino_http.so -Wl,-rpath,$ORIGIN -L/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/
#0      runProcess (package:native_toolchain_c/src/utils/run_process.dart:85:5)
<asynchronous suspension>
#1      RunCBuilder._compile (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:230:5)
<asynchronous suspension>
#2      RunCBuilder.runClangLike (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:202:7)
<asynchronous suspension>
#3      RunCBuilder.run (package:native_toolchain_c/src/cbuilder/run_cbuilder.dart:124:7)
<asynchronous suspension>
#4      CBuilder.run (package:native_toolchain_c/src/cbuilder/cbuilder.dart:205:5)
<asynchronous suspension>
#5      build (package:hooks/src/api/build_and_link.dart:191:5)
<asynchronous suspension>
#6      main (file:///Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/hook/build.dart:10:3)
<asynchronous suspension>

  stdout:
  FINER: 2026-05-27 14:43:48.889727: Using compiler /Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang from BuildInput.cCompiler.cc.
FINER: 2026-05-27 14:43:48.890566: Trying to recognize file:///Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang.
INFO: 2026-05-27 14:43:48.891102: Running `/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --version`.
FINE: 2026-05-27 14:43:49.093713: Android (14475230, +pgo, -bolt, +lto, -mlgo, based on r574158) clang version 21.0.0 (https://android.googlesource.com/toolchain/llvm-project 9f872551d3c681d06fd303b36f16ed5c274735eb)
Target: arm64-apple-darwin25.5.0
Thread model: posix
InstalledDir: /Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin

FINE: 2026-05-27 14:43:49.100225: Tool instance file:///Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang is likely Tool(Clang).
FINER: 2026-05-27 14:43:49.100511: Looking up version with --version for ToolInstance(Clang, null, file:///Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang).
INFO: 2026-05-27 14:43:49.100654: Running `/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --version`.
FINE: 2026-05-27 14:43:49.121502: Android (14475230, +pgo, -bolt, +lto, -mlgo, based on r574158) clang version 21.0.0 (https://android.googlesource.com/toolchain/llvm-project 9f872551d3c681d06fd303b36f16ed5c274735eb)
Target: arm64-apple-darwin25.5.0
Thread model: posix
InstalledDir: /Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin

FINE: 2026-05-27 14:43:49.124497: Found version for ToolInstance(Clang, 21.0.0, file:///Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang).
INFO: 2026-05-27 14:43:49.127897: Running `/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang --target=aarch64-linux-android29 --sysroot=/Users/a/Library/Android/sdk/ndk/30.0.14904198/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/ -fPIC -O3 -Wl,-z,max-page-size=16384 -fobjc-arc -DRELEASE -DNDEBUG /Users/a/.pub-cache/hosted/pub.dev/cupertino_http-3.0.0/src/native_cupertino_bindings.m -framework Foundation --shared -o /Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/libcupertino_http.so -Wl,-rpath,$ORIGIN -L/Users/a/A/PROJECTS/b/app/.dart_tool/hooks_runner/shared/cupertino_http/build/aece7ab44c/`.
SEVERE: 2026-05-27 14:43:49.201121: error: -fobjc-arc is not supported on platforms using the legacy runtime



Target dart_build failed: Error: Building native assets failed. See the logs for more details.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildStagingDebug' (registered by plugin 'dev.flutter.flutter-gradle-plugin').
> Process 'command '/Users/a/A/PROGRAMS/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 21s
Error: Gradle task assembleStagingDebug failed with exit code 1
```
Gemini 3.5 Flash (High) solution, `cupertino_http-3.0.0/hook/build.dart`:
```dart
import 'package:code_assets/code_assets.dart';
// ...
(input, output) async {
  if (input.config.buildCodeAssets) {
    final targetOS = input.config.code.targetOS;
    if (targetOS != OS.iOS && targetOS != OS.macOS) {
      return;
    }
  }
  await CBuilder.library(
    // ...
  ).run(
    input: input,
    output: output,
    // ...
  );
}
```

--- a/pkgs/cupertino_http/CHANGELOG.md
+++ b/pkgs/cupertino_http/CHANGELOG.md
@@ -1,3 +1,9 @@
+## 3.0.1
+
+* Fix a [bug](https://github.com/dart-lang/http/issues/1925) where native
+  assets were built on non-Apple platforms.
+* Support the latest semver-major versions of the native assets packages.
+
 ## 3.0.0
 
 * **BREAKING CHANGE:** Remove `shouldUseExtendedBackgroundIdleMode` from

--- a/pkgs/cupertino_http/hook/build.dart
+++ b/pkgs/cupertino_http/hook/build.dart
@@ -2,28 +2,31 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:code_assets/code_assets.dart';
 import 'package:hooks/hooks.dart';
 import 'package:logging/logging.dart';
 import 'package:native_toolchain_c/native_toolchain_c.dart';
 
 void main(List<String> args) async {
-  await build(
-    args,
-    (input, output) async =>
-        CBuilder.library(
-          name: 'cupertino_http',
-          assetName: 'src/native_cupertino_bindings.dart',
-          sources: ['src/native_cupertino_bindings.m'],
-          language: Language.objectiveC,
-          flags: ['-fobjc-arc'],
-        ).run(
-          input: input,
-          output: output,
-          logger: Logger('')
-            ..level = Level.ALL
-            ..onRecord.listen((record) {
-              print('${record.level.name}: ${record.time}: ${record.message}');
-            }),
-        ),
-  );
+  await build(args, (input, output) async {
+    if (input.config.code.targetOS != OS.iOS &&
+        input.config.code.targetOS != OS.macOS) {
+      return;
+    }
+    await CBuilder.library(
+      name: 'cupertino_http',
+      assetName: 'src/native_cupertino_bindings.dart',
+      sources: ['src/native_cupertino_bindings.m'],
+      language: Language.objectiveC,
+      flags: ['-fobjc-arc'],
+    ).run(
+      input: input,
+      output: output,
+      logger: Logger('')
+        ..level = Level.ALL
+        ..onRecord.listen((record) {
+          print('${record.level.name}: ${record.time}: ${record.message}');
+        }),
+    );
+  });
 }

--- a/pkgs/cupertino_http/pubspec.yaml
+++ b/pkgs/cupertino_http/pubspec.yaml
@@ -1,5 +1,5 @@
 name: cupertino_http
-version: 3.0.0
+version: 3.0.1
 description: >-
   A macOS/iOS Flutter plugin that provides access to the Foundation URL
   Loading System.
@@ -10,12 +10,13 @@ environment:
 
 dependencies:
   async: ^2.5.0
+  code_assets: ^1.0.0
   ffi: ^2.1.0
-  hooks: ^1.0.2
+  hooks: ^2.0.0
   http: ^1.5.0
   http_profile: ^0.1.0
   logging: ^1.3.0
-  native_toolchain_c: ^0.17.6
+  native_toolchain_c: ^0.19.1
   objective_c: ^9.1.0
   web_socket: '>=0.1.5 <2.0.0'
 

